extensions/control-plane/store/sql/contract-definition-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractdefinition/SqlContractDefinitionStore.java

```diff
@@ -153,9 +163,25 @@ private void insertInternal(Connection connection, ContractDefinition definition
                     definition.getContractPolicyId(),
                     toJson(definition.getAssetsSelector()),
                     definition.getCreatedAt());
+
+            insertProperties(definition, connection);
         });
     }
 
+    private void insertProperties(ContractDefinition definition, Connection connection) {
+        if (Objects.nonNull(definition.getPrivateProperties())) {
```

Comment:
as I wrote before, privateProperties can never be null (because they are initialized to an empty map and there's a check on the builder method, the getter is also marked with `@NotNull`)

Reply:
This is addresses